hazelcast/src/test/java/com/hazelcast/cache/CacheBasicAbstractTest.java

```diff
@@ -274,175 +272,144 @@ public void testIterator() {
 
     @Test
     public void testIteratorRemove() {
-        ICache cache = createCache();
+        ICache<Integer, Integer> cache = createCache();
         int size = 1111;
         for (int i = 0; i < size; i++) {
             cache.put(i, i);
         }
 
-        Iterator<Cache.Entry<Integer, Integer>> iter = cache.iterator();
-        while (iter.hasNext()) {
-            iter.next();
-            iter.remove();
+        Iterator<Cache.Entry<Integer, Integer>> iterator = cache.iterator();
+        while (iterator.hasNext()) {
+            iterator.next();
+            iterator.remove();
         }
         assertEquals(0, cache.size());
     }
 
     @Test(expected = IllegalStateException.class)
     public void testIteratorIllegalRemove() {
-        ICache cache = createCache();
+        ICache<Integer, Integer> cache = createCache();
         int size = 10;
         for (int i = 0; i < size; i++) {
             cache.put(i, i);
         }
 
-        Iterator<Cache.Entry<Integer, Integer>> iter = cache.iterator();
-        if (iter.hasNext()) {
-            iter.remove();
+        Iterator<Cache.Entry<Integer, Integer>> iterator = cache.iterator();
+        if (iterator.hasNext()) {
+            iterator.remove();
         }
     }
 
     @Test
-    public void testIteratorDuringInsertion() throws InterruptedException {
-        final AtomicBoolean stop = new AtomicBoolean(false);
-        final ICache cache = createCache();
-        int size = 1111;
+    @SuppressWarnings("WhileLoopReplaceableByForEach")
+    public void testIteratorDuringInsertion_withoutEviction() {
+        final ICache<Integer, Integer> cache = createCache();
+        final int size = 1111;
         for (int i = 0; i < size; i++) {
             cache.put(i, i);
         }
 
-        final Thread thread = new Thread() {
-            public void run() {
-                Random rand = new Random();
-                while (!stop.get()) {
-                    int i = rand.nextInt();
-                    try {
-                        cache.put(i, i);
-                        LockSupport.parkNanos(1);
-                    } catch (Throwable ignored) {
-                    }
-                }
+        AbstractCacheWorker worker = new AbstractCacheWorker() {
+            @Override
+            void doRun(Random random) {
+                int i = random.nextInt(size);
```

Comment:
That sounds very reasonable. I'll prepare an update of the PR on Monday. Thanks again for those insights about the eviction, that should make the changes very easy.